src/sentry/eventstream/kafka/consumer.py

```diff
@@ -0,0 +1,268 @@
+        # The commit log consumer must start consuming from the beginning of
+        # the commit log topic to ensure that it has a comprehensive view of
+        # all active partitions.
+        consumer.assign([
+            TopicPartition(
+                i.topic,
+                i.partition,
+                positions.get((i.topic, i.partition), OFFSET_BEGINNING),
+            ) for i in assignment
+        ])
+
+    consumer.subscribe(
+        [commit_log_topic],
+        on_assign=rewind_partitions_on_assignment,
+    )
+
+    while not stop_request_event.is_set():
+        message = consumer.poll(1)
+        if message is None:
+            continue
+
+        error = message.error()
+        if error is not None:
+            raise Exception(error)
+
+        positions[(message.topic(), message.partition())] = message.offset() + 1
+
+        group, topic, partition, offset = get_commit_data(message)
+        if group != synchronize_commit_group:
+            logger.debug('Received consumer offsets update from %r, ignoring...', group)
+            continue
+
+        partition_state_manager.set_remote_offset(topic, partition, offset)
+
+
+class SynchronizedConsumer(object):
+    """
+    This class implements the framework for a consumer that is intended to only
+    consume messages that have already been consumed and committed by members
+    of another consumer group.
+
+    This works similarly to the Kafka built-in ``__consumer_offsets`` topic.
+    The consumer group that is being "followed" (the one that must make
+    progress for our consumer here to make progress, identified by the
+    ``synchronize_commit_group`` constructor parameter/instance attribute) must
+    report its offsets to a topic (identified by the ``commit_log_topic``
+    constructor parameter/instance attribute). This consumer subscribes to both
+    commit log topic, as well as the topic(s) that we are actually interested
+    in consuming messages from. The messages received from the commit log topic
+    control whether or not consumption from partitions belonging to the main
+    topic is paused, resumed, or allowed to continue in its current state
+    without changes.
+
+    The furthest point in any partition that this consumer should ever consume
+    to is the maximum offset that has been recorded to the commit log topic for
+    that partition. If the offsets recorded to that topic move
+    non-monotonically (due to an intentional offset rollback, for instance)
+    this consumer *may* consume up to the highest watermark point. (The
+    implementation here tries to pause consuming from the partition as soon as
+    possible, but this makes no explicit guarantees about that behavior.)
+
+    Subclasses are required to implement the ``handle`` method which is
+    provided with a message that is ready to be processed as its only argument.
+    Implementations are also responsible for managing their own offset commits.
```

Comment:
I need to update this last paragraph because it's no longer true (this can just be used like a "normal" consumer, it supports a subset of the API)